src/crimson/osd/pg.cc

```diff
@@ -194,20 +242,30 @@ seastar::future<> PG::read_state(ceph::os::FuturizedStore* store)
 }
 
 void PG::do_peering_event(
```

Comment:
I'm not sure this is really necessary.  My other PR has a patch which addresses it by placing this under the peering pipeline stage for advance_map and pg creation.